docs/DeveloperGuide.md

```diff
@@ -196,11 +196,11 @@ How the parsing works:
 
 ![Class structure of Command](images/CommandDiagram.png)
 
-The command class structure has been changed to provide an additional layer of abstraction using the four interface
-classes ```ModelCommand```, ```StorageCommand```, ```FileCommand``` and ```MiscCommand```. These interfaces all
-implement the Command interface and is used as a intermediate barrier to build the command classes. The specific
-commands implement these commands instead of directly implementing the Command interface in order to improve
-the abstraction of commands.
+The command class structure has been changed to provide an additional layer of abstraction using the five
+classes ```ModelCommand```, ```StorageCommand```, ```FileCommand``` , ```VenueCommand```, and ```MiscCommand```. 
+These interfaces and abstract classes all implement the Command interface and is used as a intermediate barrier 
+to build the command classes. The specific commands extend or implement these command classes instead of directly implementing the 
+command interface in order to improve the abstraction of commands.
```

Comment:
Actually, we should keep the original, because `ModelCommand` is an interface, just like `StorageCommand`, `StorageModelCommand` and `MiscCommand`. Though, we do have to change `FileCommand` to `StorageModelCommand`. The idea is that each of these commands represent a type of command that executes on each component of the application, for e.g., `ModelCommand` executes on the `Model` and so on. Basically, these interfaces exists so that the `LogicManager` is able to only pass whatever is needed to the `execute` method of each command.

Though what you have mentioned is valid, so we can keep parts of it, but maybe in a different section. This part could be phrased in such a way where some commands have common logic, such as `FileCommand`, which consists of `FileCreateCommand`, `FileDeleteCommand`, `FileSwitchCommand` and `ImportCommand`, which have common logic that is abstracted in `FileCommand`. This idea is different from the idea intended in the original paragraph. For instance, while they are all `FileCommand`s, `FileCreateCommand`, `FileDeleteCommand`, `ImportCommand` all operate on only the `Storage` as they implement the `StorageCommand`, whereas the `FileSwitchCommand` operates on both `Model` and `Storage` as it implements the `StorageModelCommand`. While they implement different `Command` interfaces, they do have common logic which is abstracted in `FileCommand`.

Do ask if you have any questions about this!